govcd/api_vcd_versions.go

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2019 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"sort"
+
+	semver "github.com/hashicorp/go-version"
+
+	"github.com/vmware/go-vcloud-director/v2/util"
+)
+
+type VersionInfo struct {
+	Version    string `xml:"Version"`
+	LoginUrl   string `xml:"LoginUrl"`
+	Deprecated bool   `xml:"deprecated,attr,omitempty"`
+}
+
+type VersionInfos []VersionInfo
+
+type SupportedVersions struct {
+	VersionInfos `xml:"VersionInfo"`
+}
+
+// APIMaxVerIs allows to compare against maximum vCD supported API version. It will always
+// return false until the client has not done Authenticate(). Can be useful to validate
```

Comment:
Why do we need to authenticate first? Aren't versions returned even without authenticating?

Reply:
They are, but I wasn't willing to do additional call. This rings some bells and I will think how to make them work unauthenticated while storing this info to avoid multiple calls.

Reply:
I have changed this to work without authentication.